crates/fortitude_linter/src/rules/obsolescent/specific_names.rs

```diff
@@ -57,13 +57,28 @@ fn map_specific_intrinsic_functions(name: &str) -> Option<&'static str> {
 }
 
 /// ## What does it do?
-/// Checks for uses of the deprecated specific names of intrinsic functions.
+/// Checks for uses of the deprecated specific names of intrinsic functions, such
+/// as `ALOG` and `CLOG` for the natural logarithm of a real or complex argument
+/// respectively.
 ///
 /// ## Why is this bad?
 /// Specific names of intrinsic functions can be obscure and hinder readability of
 /// the code. Fortran 90 made these specific names redundant and recommends the use
 /// of the generic names for calling intrinsic functions.
 ///
+/// ## Examples
+/// ```f90
+/// ! Not recommended
+/// result = ALOG(x)
+///
+/// ! Better
+/// result = LOG(x)
+/// ```
+///
+/// ## Notes
+/// Specific names were officially declared obsolescent in Fortran 2018, so
+/// this rule only triggers if the target standard is Fortran 2018 or later.
```

Comment:
The generic names are possible in F90, so I don't think we should gate these, especially not as late as F2018